apps/s_client.c

```diff
@@ -798,6 +801,31 @@ static int new_session_cb(SSL *S, SSL_SESSION *sess)
     return 0;
 }
 
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    BIO *keylog = BIO_new_file(keylog_file, "a");
+    char buffer[256];
```

Comment:
If BIO can do the buffering, the missing newline should be OK for s_client, but what about other programs like curl?

@davidben WDYT about changing `ssl_log_secret` (the keylog callback) in BoringSSL to always append a newline? Every user likely wants to append a newline. Eg. from `tool/client.cc`:
```c
static void KeyLogCallback(const SSL *ssl, const char *line) {
  fprintf(g_keylog_file, "%s\n", line);
  fflush(g_keylog_file);
}
```